portal/app/farm_updater/sonnen_api.py

```diff
@@ -1,38 +1,78 @@
 import requests
+from django.conf import settings
 
 
-def _get_batteries_status_json():
-    # This method does a request to sonnen api to get status
-    serial1 = '67682'
-    # serial2 = '67670'
-    token = '5db92cf858eebce34af146974f49f4d40ec699b99372546c0af628fb48133f61'
-    url_ini = 'https://core-api.sonnenbatterie.de/proxy/'
-    headers = {'Accept': 'application/vnd.sonnenbatterie.api.core.v1+json', 'Authorization': 'Bearer ' + token}
-    status_endpoint = '/api/v1/status'
 
-    try:
-        resp = requests.get(url_ini + serial1 + status_endpoint, headers=headers)
-        resp.raise_for_status()
-        data = resp.json()
-        data['batt_id'] = serial1
-        return data
+class SonnenApiInterface:
 
-    except requests.exceptions.HTTPError as err:
-        print('Error get_battery_status_json')
-        return None
+    def __init__(self, url=settings.SONNEN_URL, token=settings.SONNEN_TOKEN):
+        self.url = url
+        self.token = token
+        self.headers = {'Accept': 'application/vnd.sonnenbatterie.api.core.v1+json', 'Authorization': 'Bearer ' +
+                                                                                                      self.token}
+
+    def get_batteries_status_json(self, serial):
+        # This method does a request to sonnen api to get status
+        status_endpoint = '/api/v1/status'
+
+        try:
+            resp = requests.get(self.url + serial + status_endpoint, headers=self.headers)
+            resp.raise_for_status()
+            data = resp.json()
+            data['batt_id'] = serial
+            return data
+
+        except requests.exceptions.HTTPError as err:
+            print('Error get_battery_status_json')
+            return None
+
+    def manual_mode_control(self, serial, mode='charge', value='0'):
+        control_endpoint = '/api/v1/setpoint/'
+        # Checking if system is in off-grid mode
+        voltage = self.get_batteries_status_json().get_status()['Uac']
+
+        if voltage == 0:
+            print('Battery is in off-grid mode... Cannot execute the command')
+            return {}
+
+        else:
```

Comment:
Because of the return above, this else is unnecessary. It simplifies reading the flow.